src/main/java/io/vertx/core/http/impl/HttpServerImpl.java

```diff
@@ -1019,5 +1063,11 @@ private void http1(ChannelHandlerContext ctx, ByteBuf buf) {
       ctx.fireChannelRead(buf);
       ctx.pipeline().remove(this);
     }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+      ctx.channel().close();
+      HttpServerImpl.this.connectionExceptionHandler.handle(cause);
```

Comment:
PTAL

2016-09-13 18:24 GMT+02:00 Julien Viet <notifications@github.com>:

> In src/main/java/io/vertx/core/http/impl/HttpServerImpl.java
> <https://github.com/eclipse/vert.x/pull/1620#discussion_r78594371>:
>
> > @@ -1019,5 +1063,11 @@ private void http1(ChannelHandlerContext ctx, ByteBuf buf) {
> >        ctx.fireChannelRead(buf);
> >        ctx.pipeline().remove(this);
> >      }
> > +
> > +    @Override
> > +    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
> > +      ctx.channel().close();
> > +      HttpServerImpl.this.connectionExceptionHandler.handle(cause);
>
> I think it should use Context#executeFromIO and also test handler is not
> null.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/vert.x/pull/1620/files/4fefecfc7172bb280bd4f7979311570441a8baf0#r78594371>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABblth4ZytZf24vUblNjnTyfYzT2_dIwks5qps4sgaJpZM4J44vk>
> .
>